pkg/commands/archive.go

```diff
@@ -1,29 +1,23 @@
 package commands
 
 import (
-	"archive/tar"
-	"bufio"
-	"bytes"
-	"compress/gzip"
 	"fmt"
-	"io"
+	"io/ioutil"
 	"net/http"
 	"os"
-	"path/filepath"
 
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/fileutils"
 	"github.com/sirupsen/logrus"
 
 	build "github.com/Azure/acr-builder/pkg"
 	"github.com/Azure/acr-builder/pkg/constants"
 )
 
-const maxHeaderSize = 4
-
-// NOTE: only support .tar.gz for now
-var supportedArchiveHeaders = map[byte][]byte{
-	// Gzip
-	0x1F: {0x1F, 0x8B, 0x08},
-}
+const (
+	tempWorkingDir = "temp"
+	tempFileName   = "temp.tar.gz"
+)
 
 // ArchiveSource defines source in the form of an archive file
 // Currently we only support tar.gz
```

Comment:
Do we still have the limitation?

Reply:
Na, I just named it that for clarity and if you use software to open it you can use it automatically. The extension doesn't do anything. We can use bzip2 and stuff if we want, I haven't tested all the formats though.